scripts/validateBuild.ts

```diff
@@ -0,0 +1,48 @@
+import { readFileSync, readdirSync } from 'fs'
+
+export interface ExpectedBuildOutput {
+  buildDir: string
+  files: string[]
+  clientOnlyModules: string[]
+}
+
+export class MissingBuildFileError extends Error {
+  override name = 'MissingBuildFileError'
+  constructor(
+    public cause: Error,
+  ) {
+    super(cause.message, { cause })
+  }
+}
+
+export class MissingClientOnlyDirectiveError extends Error {
+  override name = 'MissingClientOnlyDirectiveError'
+  constructor(
+    public cause: Error,
+  ) {
+    super(cause.message, { cause })
+  }
+}
+
+function validateClientOnlyBuildFile(filePath: string) {
+  const fileContent = readFileSync(filePath, 'utf8')
+  if (!fileContent.startsWith('"use client"')) {
```

Comment:
Just to be thorough, this should also check that the semicolon is present following the directive otherwise it could mean that the minified bundle will not be able to be parsed